### ISSUE TITLE
ejabberd - blog post comments disabled

### ISSUE THREAD

**Issue by ghost** [movim/movim#149]
Posting new blog posts is fine, but comments are disabled, and I cannot find any way to enable them.

Metronome works fine, so this leads me to believe it's something wrong with the ejabberd configuration.

![capture](https://cloud.githubusercontent.com/assets/1242720/13621021/2f7fdbb8-e5f9-11e5-8b69-96f029eb6bde.PNG)

I understand if this is a "won't fix", I just thought it would be a good idea to bring it up in case someone else has a similar issue.

**Comment by edhelas** [comment]
No this is definitely a bug. I have now access to a fresh ejabberd server so I'll be able to debug :)

**Comment by ghost** [comment]
Ok, I found a couple of things that may be related.  Would you like me to post more details?

One is basically not being able to subscribe to a PubSub node (that you didn't create) on an ejabberd server.

The other is not being able to save your vcard on an ejabberd server.

They seem to work fine on other servers like Metronome, but not ejabberd.

I'm wondering if it has to do with DNS entries...

I'll keep testing as well to see if i can narrow it down.

**Comment by edhelas** [comment]
Ok thanks, for each issue that you found please open a new bug with a precise way to reproduce it ;)

**Comment by inkhey** [comment]
I got the same issue with comments on fresh install of prosody and sat-pubsub.
It's maybe the same problem.

**Comment by alexandrelb** [comment]
inkhey : I have the same install (prosody 0.10 & sat_pubsub 0.2) I think this come from sat_pubsub, I open a bug here : http://bugs.goffi.org/show_bug.cgi?id=141

**Comment by Trim** [comment]
Hello,

I have an ejabberd server too and I've seen the same behavior.

I wanted to add that I've created an "IPv6" group on my ejabberd server to test it.
It seems that I'm able to add a comment for a post inside this group:

![screenshot of successfull comment](https://cloud.adorsaz.ch/index.php/s/dluYRty7HBDFro0/download)

My ejabberd pubsub service use a PostgreSQL database as storage.

I can check the `urn:xmpp:microblog:0` node configuration with Gajim 0.16, but I have no idea which configuration is needed to allow comment on microblog posts.

**Comment by Trim** [comment]
Ah, I've tried to apply the group configuration to my microblog configuration using Gajim.

I've seen in ejabberd log this error:

```
2016-07-29 15:29:49.014 [error] <0.318.0>@ejabberd_sql:sql_query_internal:532 PREPARE failed for SQL query at {nodetree_tree_sql,79}: [{severity,'ERREUR'},{code,<<"42601">>},{message,<<"erreur de syntaxe sur ou près de « node »">>},{position,32},{file,<<"scan.l">>},{line,1052},{routine,<<"scanner_yyerror">>}]
```

That's certainly related (the french message indicates a syntax error in the SQL query)

PS: In this configuration the "number of maximum elements to save" is configured to 1 which can explain the "Comments disabled".

**Comment by Trim** [comment]
I've applied the patch from processone/ejabberd#1219 on my server and it seems to work fine now (I can comment on private and public posts).

@starblessed Does your ejabberd configuration have a SQL storage for pubsub section different from the default one (Mnesia data base) ?

If that's the case, the ejabberd patch should be enough to fix this bug on Movim.

**Comment by edhelas** [comment]
Because it was a ejabberd issue and the bug is solved on their side. I'm closing this one :)

**Comment by p5n** [comment]
I got the same issue with
- ejabberd 17.08-1
- movim-git r5728.d54bb492-1
and mnesia database (only mod_mam database is in mysql)

**Comment by rpallai** [comment]
I got a related issue: "comments disabled" on posts without like. Posts with one like at least are commentable. Weird. ejabberd 18.01

**Comment by c-93** [comment]
I have the same problem. #625. Posts with pictures have comments enabled only after at least one like.
